src/components/TradeWidget/index.tsx

```diff
@@ -391,7 +391,12 @@ const TradeWidget: React.FC = () => {
   const { balances, tokens: tokenList } = useTokenBalances()
 
   // If user is connected, use balances, otherwise get the default list
-  const tokens = isConnected && balances.length > 0 ? balances : tokenList
+  const tokens =
```

Comment:
Would this logic make more sense inside the hook?

Reply:
no, this is specific to TradeWidget, because inside useTokenBalance we need to have empty array so we don't check balances for wrong addresses
But TradeWidget gets `[sell, receive] = tokenList` and then counts on them being defined

Unless you want a new specific hook only for that one line

Reply:
I DO NOT